### PR TITLE
add function and relevant packages

### PR DESCRIPTION
**Pull request by JahanzaibNaveed**
Add a function to get all the branches being used in a teamcity configuration

### CODE REVIEW

File: TeamcityRestApi/TeamcityConnector.fs [hunk 1]
```diff
@@ -825,6 +828,18 @@ type TeamcityConnector(httpconnector : IHttpTeamcityConnector) =
             let payload = sprintf """<builds><build id="%s"/></builds>""" buildid
             let requestDAta = httpconnector.HttpPutXmlContent(conf, url, payload)
             requestDAta.IsSuccessful
+        
+        member this.GetBranchesFromTeamcityConfiguration(conf: ITeamcityConfiguration, buildTypeId: string) =
+            let url = sprintf "app/rest/buildTypes/id:%s/branches" buildTypeId
+            let content = httpconnector.HttpRequest(conf, url, RestSharp.Method.Get).Content
+            let branches = new System.Collections.Generic.List<string>()
```
Comment (jmecosta) [inline TeamcityRestApi/TeamcityConnector.fs:835]: u dont need to use System.Collections.Generic.List<string>, u can use generic FSharp lists, see how other functions are done

File: TeamcityRestApi/TeamcityConnector.fs [hunk 2]
```diff
@@ -825,6 +828,18 @@ type TeamcityConnector(httpconnector : IHttpTeamcityConnector) =
             let payload = sprintf """<builds><build id="%s"/></builds>""" buildid
             let requestDAta = httpconnector.HttpPutXmlContent(conf, url, payload)
             requestDAta.IsSuccessful
+        
+        member this.GetBranchesFromTeamcityConfiguration(conf: ITeamcityConfiguration, buildTypeId: string) =
+            let url = sprintf "app/rest/buildTypes/id:%s/branches" buildTypeId
+            let content = httpconnector.HttpRequest(conf, url, RestSharp.Method.Get).Content
```
Comment (jmecosta) [inline TeamcityRestApi/TeamcityConnector.fs:834]: you can use type providers directly to access the data.. see how the other fucntions are done

File: TeamcityRestConnect.Test/TeamcityRestConnect.Test.csproj [hunk 3]
```diff
@@ -10,6 +10,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
     <PackageReference Include="NUnit" Version="3.11.0" />
     <PackageReference Include="NUnit3TestAdapter" Version="3.13.0" />
+    <PackageReference Include="Remote.Linq.Newtonsoft.Json" Version="7.2.1" />
```
Comment (jmecosta) [inline TeamcityRestConnect.Test/TeamcityRestConnect.Test.csproj:13]: this is not required, the library has type providers to acccess the data

**Review by jmecosta — CHANGES_REQUESTED** [review]

File: TeamcityRestApi/TeamcityConnector.fs (hunk 1 above)
Comment (JahanzaibNaveed) [reply TeamcityRestApi/TeamcityConnector.fs:835]: All other functions also use System.Collections.Generic.List,

Comment (jmecosta) [reply TeamcityRestApi/TeamcityConnector.fs:835]: ah ok, u can keep then

**Review by jmecosta — APPROVED** [review]

**Pull request merged**